### ISSUE TITLE
sound: Write tests for sound.c functions

### ISSUE THREAD

**Issue by jiaoboxuan** [uchicago-cs/chiventure#1600]
In this issue, we need to write tests for `sound.c` functions, which include `load_wav`, `sound_free`, and `play_sound`. These tests will allow us to make sure that all the functions are working properly, so that we can make a successful demo.

Assignees: Rhea, Victor, Jake

**Comment by sangerr** [comment]
Now that we have the sound_new and sound_init functions, these can help us in writing tests for the functions without having to repeat code. We are running into some issues with testing most of the functions due to Cmake and compiler errors. Ultimately, we may have to get rid of 6/8 tests we wrote.

**Comment by jwschroeder21** [comment]
Update, Elizabeth noted how these sounds are the same ones we added in another part of the code. Now, instead of duplicating the files, we will reference the file path of where they already exist.

**Comment by sangerr** [comment]
Closing Statement: In this issue, we wrote tests for sound.c functions. While we initially wrote tests for sound_new, sound_init, load_wav, sound_free, and play_sound, there were large issues we encountered in trying to compile them. We ultimately only tested sound_new, which is the most critical function to sound.c, and were able to have those tests compile and pass. We would urge future teams to tackle the other tests, as we have written the implementation for all of them, but were unable to commit it.

Note: We made the changes requested by Elizabeth and will close this issue as soon as the PR is merged
